.github/workflows/publish-gem.yaml

```diff
@@ -24,6 +24,28 @@ jobs:
       - name: Install dependencies
         run: bundle install
 
+      - name: Bump version
+        run: |
+          NEW_VERSION=$(echo ${{ github.event.release.tag_name }} | sed 's/^v//') # strip the 'v' from the tag if present
+          sed -i -r "s/VERSION = \".+\"/^VERSION = \"${NEW_VERSION}\"/" ./lib/descope/version.rb
+
+      - name: Commit changes
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+          git commit -am "Bump version to $NEW_VERSION"
+          git push
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Repoint the tag to latest commit
+        run: |
+          git tag -fa ${{ github.event.release.tag_name }} -m "Release $NEW_VERSION"
+          git push origin :${{ github.event.release.tag_name }}
```

Comment:
Tag is not a branch, you cannot push commits to a tag. 
As far as i understand, to release a new version you need to :
1. Create a "bump version file" PR
2. Bump
3. Approve and merge
4. Tag main with new version
5. Github Release

Reply:
the flow is:

Yael is creating a new tag and release

the release is bumping the version.rb with that tag

now commit that change

then repoint the tag to that commit

Reply:
`git push origin :${{ github.event.release.tag_name }}` is DELETING the tag so it can retag

Reply:
so this code

```yaml
git tag -fa ${{ github.event.release.tag_name }} -m "Release $NEW_VERSION"
git push origin :${{ github.event.release.tag_name }}
git push
```

is force create tag on the new commit from that bump
then delete the tag from remote
then push the changes which will push the tag that is now pointing to the code with the new version.rb
all of this happens only on release flow